evaluation.py

```diff
@@ -237,7 +237,143 @@ def predict_keypoints(self,generator, location):
                 image_ids.append(metadata['src_set_image_id'])
                 j+=1
         self.write_to_json_file(list_of_predictions, location)
-        return image_ids
+        return image_ids, list_of_predictions
+
+    # This function evaluates PCK@0.2 == Distance between predicted and true joint < 0.2 * torso diameter
+    # The PCK_THRESHOLD constant can be updated to adjust this threshold
+    # https://github.com/cbsudux/Human-Pose-Estimation-101#percentage-of-correct-key-points---pck
+    def pck_eval(self, generator, location, annotation_file):
+        _, list_of_predictions = self.predict_keypoints(generator, location)
+        f = open(annotation_file)
+        data = json.load(f)
+
+        # This function depends on the keypoints order listed in constants COCO_KEYPOINT_LABEL_ARR
+        dist_list = []
+        correct_keypoints = {
+            "nose": 0,
+            "left_eye": 0,
+            "right_eye": 0,
+            "left_ear": 0,
+            "right_ear": 0,
+            "left_shoulder": 0,
+            "right_shoulder": 0,
+            "left_elbow": 0,
+            "right_elbow": 0,
+            "left_wrist": 0,
+            "right_wrist": 0,
+            "left_hip": 0,
+            "right_hip": 0,
+            "left_knee": 0,
+            "right_knee": 0,
+            "left_ankle": 0,
+            "right_ankle": 0
+        }
+        for prediction in list_of_predictions:
+            prediction_image_id = prediction['image_id']
+            prediction_keypoints = prediction['keypoints']
+            for i, ann in enumerate(data['annotations']):
+                if data['annotations'][i]['image_id'] == prediction_image_id:
+                    annotation_keypoints = data['annotations'][i]['keypoints']
+                    prediction_keypoints = np.array(prediction_keypoints)
+                    annotation_keypoints = np.array(annotation_keypoints)
+
+                    # Calculate PCK@0.2 threshold for image
+                    # Joint at 11 is left hip, Joint at 12 is right hip. Multiply by 3 as each keypoint has (x, y, visibility) to get the array index
+                    left_hip_point = np.array(annotation_keypoints[33], annotation_keypoints[34])
+                    right_hip_point = np.array(annotation_keypoints[36], annotation_keypoints[37])
+                    torso = np.linalg.norm(left_hip_point-right_hip_point)
+                    threshold = PCK_THRESHOLD*torso
+
+                    for i in range(NUM_COCO_KEYPOINTS):
+                        base = i*3
```

Comment:
just realized we have a `const` for this
```suggestion
                        base = i * NUM_COCO_KP_ATTRBS
```